sherpa/models/regrid.py

```diff
@@ -273,29 +415,41 @@ class EvaluationSpace2D():
     ----------
     x, y : array_like or None, optional
         The data array, or the low end of the x and y data bins if the dataset
-        is "integrated".
+        is "integrated". These are not required to be the same length.
     xhi, yhi : array_like or None, optional
         The high end of the x and y data bins for integrated datasets.
     """
+
     def __init__(self, x=None, y=None, xhi=None, yhi=None):
         # In the 2D case the arrays are redundant, as they are flattened from a meshgrid.
         # We need to clean them up first to have proper axes.
         # This may happen when an EvaluationSpace2D is instantiated using the arrays passed to
         # the calc method.
+        #
+        # This means that this class does not check that x and y (if set) have
+        # the same length.
+        #
```

Comment:
This code looks well for regular grids, but I have concerns on irregular grids. 

Imagine the following integrated grid:
```
x0= [1, 2, 3, 1, 2, 3]
x1 = [1, .5, 1, 2, 2, 2]
```
What does that even mean? It really needs a spec of the interpolation method when evaluating the model; it's even worse in n-D. I'm not sure we have thought through all potential problems here. Should there be a requirement that integrated data is on a regular grid?

Reply:
I do not understand the 2D regriud code - e.g. see #840. It is being used (or was used?) for the CSC so we have to be careful not to break things, but I really just have not had the time, energy, or implication to try and resolve any of this.

Reply:
Does the `regrid` code deal with model on a higher resolution grid than the input data?

Reply:
@anetasie - I don't understand your question?